### ISSUE TITLE
Make the confirmable devise module optional

### ISSUE THREAD

**Issue by mconf-daileon** [mconf/mconf-web#531]
---

Author Name: **Leonardo Daronco** (Leonardo Daronco)
Original Redmine Issue: 1060, http://dev.mconf.org/redmine/issues/1060

---

Currently when users register a new account they are required to confirm their email account. A notification will be shown in the website until the email is confirmed.

Now we have an option to require administrators to approve new accounts (see #953). If this option is enabled, it will, in some cases, be useless to require the users to also confirm their emails. Once an administrator approved the account, he's in some way already confirming the user's email.

So we should make the module 'confirmable' from devise optional. A flag ("require email confirmation", for example) should be added to the model Site so an admin can enable or disable it. By default it should be enabled, so the site behaves as it does today. If the flag is disabled, however, users should be automatically confirmed and no confirmation email should be sent.

**Comment by mconf-daileon** [comment]
---

Original Redmine Comment
Author Name: **Leonardo Daronco** (Leonardo Daronco)
Original Date: 2014-08-20T21:01:50Z

---

What actually makes more sense is to automatically disable the confirmations if the option "require approval" is set.
After #1322, the emails will be informing the users correctly if they have to confirm their account or just wait for an approval.

The implementation could be:
* When a user is approved, he's automatically confirmed (I guess this is already done today);
* The confirmation module doesn't really need to be disabled. It can be enabled but bypassed by automatically confirming the users when necessary.

So I'm leaving this issue open to review in the future, but it should be implemented as other related tickets are implemented.
